paasta_tools/chronos_rerun.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python
+# Copyright 2015 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Chronos Rerun: Designed to 'rerun' jobs with the right parameters.
+
+``chronos_rerun`` will clone any given job, and give it a schedule to
+run as soon as possible, and only once. If the job being rerun is a
+'dependent job', that is, a job triggered by the successful running of 'parent'
+jobs, then it is cloned without any children attached, and run as a regular
+scheduled job.
+"""
+import argparse
+import copy
+import datetime
+
+import service_configuration_lib
+
+from paasta_tools import chronos_tools
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import NoDeploymentsAvailable
+from paasta_tools.utils import NoDockerImageError
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='',
+    )
+    parser.add_argument('-v', '--verbose', action='store_true', dest="verbose", default=False,
+                        help="Print out more output regarding the state of the service")
+    parser.add_argument('-d', '--soa-dir', dest="soa_dir", metavar="SOA_DIR",
+                        default=service_configuration_lib.DEFAULT_SOA_DIR,
+                        help="define a different soa config directory")
+    parser.add_argument('service_instance', help='Instance to operate on. Eg: example_service.main')
+    parser.add_argument('execution_date', help="The date the job should be rerun for")
```

Comment:
I understand this is the backend part, but should we mention which date format we expect, here? (`%Y-%m-%dT%H:%M:%S`)